sys/include/xtimer.h

```diff
@@ -429,15 +429,17 @@ static inline bool xtimer_less64(xtimer_ticks64_t a, xtimer_ticks64_t b);
 /**
  * @brief lock a mutex but with timeout
  *
- * @note this requires core_thread_flags to be enabled
+ * @note if the timeout is lower than XTIMER_BACKOFF it is only tested if
+ * the mutex can be locked.
  *
- * @param[in]    mutex  mutex to lock
- * @param[in]    us     timeout in microseconds relative
+ * @param[in]    mutex    mutex to lock
+ * @param[in]    timeout  timeout in microseconds relative
  *
  * @return       0, when returned after mutex was locked
- * @return       -1, when the timeout occcured
+ * @return       -1, mutex can't be locked right now (timeout <= XTIMER_BACKOFF)
+ * @return       -2, when the timeout occcured
```

Comment:
Can you introduce an enum for the error values? This would make it easier to change this later on.